File: src/main/java/com.shangguigu.chapter07/AnonymousSubClass.java
package com.shangguigu.chapter07;

/**
 * Created by liuquan on 2020/3/5.
 */
public class AnonymousSubClass {
    public static void main(String[] args) {
        A2 a2 = new A2() {
            public void cry() {
                System.out.println("cry ...");
            }
        };
        a2.cry();
    }
}

abstract class A2 {
    abstract public void cry();
}

File: src/main/scala/com/shangguigu/chapter07/AnonymousSubClass.scala
package com.shangguigu.chapter07

/**
 * Created by liuquan on 2020/3/5.
 */
object ScalaAnonymousSubClass {
    def main(args: Array[String]): Unit = {
        val a2 = new ScalaA2 {
            def cry(): Unit = {
                println("cry ...")
            }

            var name: String = _
        }
        a2.cry()
    }
}

abstract class ScalaA2 {
    var name: String

    def cry()
}